frontend/src/store/index.js

```diff
@@ -156,6 +156,13 @@ const isValidRegion = (getters, cloudProfileName, cloudProviderKind) => {
       // Azure regions may not be zoned, need to filter these out for the dashboard
       return !!getters.zonesByCloudProfileNameAndRegion({ cloudProfileName, region }).length
     }
+
+    // Filter regions that are not defined in cloud profile
+    const cloudProfile = getters.cloudProfileByName(cloudProfileName)
+    if (cloudProfile) {
+      return !!(find(cloudProfile.data.regions, { name: region }))
```

Comment:
we usually do the following (however I haven't tested it), also there was an unnecessary bracket
```suggestion
      return !!find(cloudProfile.data.regions, ['name', region])
```